libultraship/libultraship/Lib/Fast3D/gfx_pc.cpp

```diff
@@ -1077,19 +1063,17 @@ static void gfx_sp_vertex(size_t n_vertices, size_t dest_index, const Vtx *verti
                 dotx /= 127.0f;
                 doty /= 127.0f;
 
-                if (dotx < -1.0f) dotx = -1.0f;
-                if (dotx > 1.0f) dotx = 1.0f;
-                if (doty < -1.0f) doty = -1.0f;
-                if (doty > 1.0f) doty = 1.0f;
+                std::clamp(dotx, -1.0f, 1.0f);
+                std::clamp(doty, -1.0f, 1.0f);
 
                 if (rsp.geometry_mode & G_TEXTURE_GEN_LINEAR) {
                                     // Not sure exactly what formula we should use to get accurate values
                                     /*dotx = (2.906921f * dotx * dotx + 1.36114f) * dotx;
                                     doty = (2.906921f * doty * doty + 1.36114f) * doty;
                                     dotx = (dotx + 1.0f) / 4.0f;
                                     doty = (doty + 1.0f) / 4.0f;*/
-                dotx = acosf(-dotx) /*/ (3.14159265f)*/ / 4.0f;
-                doty = acosf(-doty) /*/ (3.14159265f)*/ / 4.0f;
```

Comment:
The `M_PI` was commented out and should not be introduced.

Reply:
great catch! fixed this now :)